### PR TITLE
Feature/beta distribution metric

### PR DESCRIPTION
**Pull request by JakeForsey**
Attempt at implementing Beta distribution for data in range [0, 1].

I could use some thoughts on:
 * Changes to GroupNormalizer and TorchNormalizer
 * New Normalizer

NOTE: Test is not yet finished (need a normalizer that makes sense....)

### CODE REVIEW

**Comment by jdb78** [comment]
Couple of thoughts here:
* It might make sense to reparametrize as by mean and shape beta as α = μν, β = (1 − μ)ν where the original concentrations are parameters α and β.
* Normalization should probably be a deterministic/not-learned transformation to scale data to 0-1. As the beta distribution as bounds, the user should know upfront how to normalize. This is could be done before feeding to the timeseries dataset by the user. Then, you can use the standard target normalizers - no need to modify them.
* Finally, using the target scale (mean, std), in the metric's `rescale_parameters()` method, you could try μ_new= sigmoid(μ + mean) and ν_new = softplus(ν) * μ_new * (1- μ_new) / std**2  (just a quick suggestion, there might be better ways to to calculate μ_new and ν_new). From  μ_new and ν_new, you can calculate α and β to create the distribution from which you can sample.
As always, [the wikipedia article on the Beta distribution](https://en.wikipedia.org/wiki/Beta_distribution) is helpful for reparametrizations.

I hope this is helpful!

**Comment by JakeForsey** [comment]
I've done the re-parameterisation and my model is training well. 

I'm not sure how to write a proper tests for it, in a bit over my head.

File: pytorch_forecasting/metrics.py [hunk 1]
```diff
@@ -835,3 +835,37 @@ def rescale_parameters(
...
+    def rescale_parameters(
+        self, parameters: torch.Tensor, target_scale: torch.Tensor, transformer: BaseEstimator
+    ) -> torch.Tensor:
+        assert transformer.coerce_positive, "Beta distribution is only compatible with strictly positive data"
+        assert (
+            not transformer.log_scale
+        ), "Beta distribution is not compatible with log transformation - use LogNormal"
+        assert not transformer.center, "Beta distribution is not compatible with centered data"
+
+        scaled_mean = torch.sigmoid(parameters[..., 0] + target_scale[..., 0].unsqueeze(1))
```
Comment (jdb78) [inline pytorch_forecasting/metrics.py]: I wonder if it works even better if you take `torch.logit(target_scale[..., 0])` instead of `target_scale[..., 0]`.

File: tests/test_metrics.py [hunk 2]
```diff
@@ -14,6 +14,7 @@
     LogNormalDistributionLoss,
     NegativeBinomialDistributionLoss,
     NormalDistributionLoss,
+    BetaDistributionLoss,
```
Comment (jdb78) [inline tests/test_metrics.py]: Could you run black and isort (best is to use the pre-commit hooks to ensure correct formatting)?

**Review by jdb78 — CHANGES_REQUESTED** [review]
Looks really good! Thanks a lot for all the hard work! Just minimal formatting changes and a suggestion on the parametrization. How are you training? Maybe you could write some documentation as to how it can be used by a user and what they should take consider.

**Comment by jdb78** [comment]
Integrated the beta distribution loss - should work now

**Review by jdb78 — APPROVED** [review]

**Comment by JakeForsey** [comment]
This is awesome, much better than what I was messing around with.
